python/lsst/meas/extensions/scarlet/io.py

```diff
@@ -472,3 +482,122 @@ def oldScarletToData(blend: Blend, psfCenter: tuple[int, int], xy0: Point2I):
     )
 
     return blendData
+
+
+class ScarletModelFormatter(TypelessFormatter):
+    """Read and write zip archives.
+
+    In order for files to be read from a zip file, the pydantic model
+    must have a `load` method that accepts a file object and the filename
+    as arguments, as well as a `from_zip` method that accepts a dictionary
+    of files to create the object from the zip archive.
+    """
+
+    default_extension = ".scarlet"
+    unsupported_parameters = None
```

Comment:
I think you might want this to be `frozenset()` rather than `None`. Are you sure parameters are being passed in and it's not using your storage class delegate to filter after the fact?